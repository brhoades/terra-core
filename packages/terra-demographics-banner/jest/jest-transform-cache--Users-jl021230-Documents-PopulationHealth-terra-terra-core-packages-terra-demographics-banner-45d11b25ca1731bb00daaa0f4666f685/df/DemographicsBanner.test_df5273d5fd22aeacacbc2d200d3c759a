/* istanbul ignore next */'use strict';var /* istanbul ignore next */_react = require('react'); /* istanbul ignore next */var _react2 = _interopRequireDefault(_react);
var /* istanbul ignore next */_DemographicsBanner = require('../../src/DemographicsBanner'); /* istanbul ignore next */var _DemographicsBanner2 = _interopRequireDefault(_DemographicsBanner);function _interopRequireDefault(obj) {return obj && obj.__esModule ? obj : { default: obj };}

describe('DemographicsBanner', function () {
  it('renders a blank banner properly', function () {
    var banner = shallow( /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_DemographicsBanner2.default, /* istanbul ignore next */null));
    expect(banner).toMatchSnapshot();
  });

  it('renders a banner that has basic information', function () {
    var banner = shallow(
    /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_DemographicsBanner2.default, /* istanbul ignore next */{
      age: '24',
      gender: 'F',
      personName: 'Terra Demographics Banner',
      dateOfBirth: 'March 30, 2017' }));



    expect(banner).toMatchSnapshot();
  });

  it('renders the banner that contains additional information', function () {
    var banner = shallow(
    /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_DemographicsBanner2.default, /* istanbul ignore next */{
      photo: /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'img', /* istanbul ignore next */{ alt: 'My Cat', src: 'http://lorempixel.com/100/100/animals/7/' }),
      personName: 'Johnathon Doe',
      preferredFirstName: 'John',
      gender: 'Male',
      age: '25 Years',
      dateOfBirth: 'May 9, 1993',
      additionalDetails: /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'span', /* istanbul ignore next */{ className: 'risk-score' }, /* istanbul ignore next */'5%'),
      identifiers: { MRN: 12343, REA: '3JSDA' },
      applicationRows: [{ Address: '1501 Walnut St. Apt #123', City: 'Kansas City MO' }, { ZIP: 64108 }] }));



    expect(banner).toMatchSnapshot();
  });

  it('renders the banner appropriately for a deceased person', function () {
    var banner = shallow(
    /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_DemographicsBanner2.default, /* istanbul ignore next */{
      age: '24',
      gender: 'F',
      personName: 'Old Terra Demographics Banner',
      dateOfBirth: 'June 15, 2014',
      deceasedDate: 'March 30, 2017' }));



    expect(banner).toMatchSnapshot();
  });

  it('renders the banner properly for a deceased person with additional application content', function () {
    var banner = shallow(
    /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */_DemographicsBanner2.default, /* istanbul ignore next */{
      photo: /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'img', /* istanbul ignore next */{ alt: 'My Cat', src: 'http://lorempixel.com/100/100/animals/7/' }),
      personName: 'Johnathon Doe',
      preferredFirstName: 'John',
      gender: 'Male',
      age: '25 Years',
      deceased: 'March 5, 2016',
      dateOfBirth: 'May 9, 1993',
      additionalDetails: /* istanbul ignore next */_react2.default.createElement( /* istanbul ignore next */'span', /* istanbul ignore next */{ className: 'risk-score' }, /* istanbul ignore next */'5%'),
      identifiers: { MRN: 12343, REA: '3JSDA' },
      applicationRows: [{ Address: '1501 Walnut St. Apt #123', City: 'Kansas City MO' }, { ZIP: 64108 }] }));



    expect(banner).toMatchSnapshot();
  });
});