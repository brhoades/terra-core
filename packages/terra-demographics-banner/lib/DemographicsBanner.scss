/* stylelint-disable  */

@import '~terra-mixins';
@import './variables';
@import './mixins';

%terra-demographics-banner {
  background: $demographics-background;
  color: #fff;
  display: -ms-flex;
  display: flex;
  margin-bottom: 10px;
  padding: 10px;

  h1 {
    margin: 0;

    .terra-DemographicsBanner--preferred-first-name {
      margin: auto 5px;

      &::before,
      &::after {
        content: '"';
      }
    }
  }
}

.terra-DemographicsBanner {
  @extend %terra-demographics-banner;
}

.terra-DemographicsBanner--deceased {
  @extend %terra-demographics-banner;
  background: $demographics-background-deceased;
}

.terra-DemographicsBanner--row {
  display: flex;

  span + span {
    display: inline-block;
    margin-left: 5px;
  }
}

.terra-DemographicsBanner--content {
  flex-grow: 1;
  margin: auto 10px;
}

.terra-DemographicsBanner--application-content,
.terra-DemographicsBanner--identifiers {
  margin-left: auto;
}

[dir='rtl'] {
  .terra-DemographicsBanner--row {
    span + span {
      margin-left: 0;
      margin-right: 5px;
    }
  }

  .terra-DemographicsBanner--application-content,
  .terra-DemographicsBanner--identifiers {
    margin-left: inherit;
    margin-right: auto;
  }
}

