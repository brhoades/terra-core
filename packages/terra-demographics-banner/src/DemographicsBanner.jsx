/* eslint-disable global-require, import/no-dynamic-require */
import React, { PropTypes } from 'react';
import i18nLoader from 'terra-i18n/lib/i18nLoader';
import I18nProvider from 'terra-i18n/lib/I18nProvider';

import './DemographicsBanner.scss';
import DemographicsBannerDisplay from './DemographicsBannerDisplay';

const propTypes = {
  /**
   * Full Name of the person
   */
  personName: PropTypes.string,
  /**
   * The persons preferred first name if they have one.
   */
  preferredFirstName: PropTypes.string,
  /**
   * Photo to display in the banner
   */
  photo: PropTypes.node,
  /**
   * Age of the person.
   */
  age: PropTypes.string,
  /**
   * Gender of the Person
   */
  gender: PropTypes.string,
  /**
   * Locale to display the banner with
   */
  locale: PropTypes.string,
  /**
   * Additional key value identifiers of a person's demographic information
   */
  identifiers: PropTypes.object,
  /**
   * The persons date of birth
   */
  dateOfBirth: PropTypes.string,
  /**
   * The persons gestational age.
   */
  gestationalAge: PropTypes.string,
  /**
   * The column layout in which to present the displays.
   */
  postMenstrualAge: PropTypes.string,
  /**
   * The person's deceased date. Will display the banner as deceased if this value is provided
   */
  deceasedDate: PropTypes.string,
  /**
   * Additional Details to display in the banner.
   */
  additionalDetails: PropTypes.node,
  /**
   * Additional rows of key value pairs to add to the demographics banner. The rows will be added after the person rows.
   */
  applicationRows: PropTypes.array,
};

const defaultProps = {
  personName: null,
  preferredFirstName: null,
  photo: null,
  age: null,
  gender: null,
  identifiers: {},
  locale: 'en',
  dateOfBirth: null,
  gestationalAge: null,
  postMenstrualAge: null,
  deceasedDate: null,
  additionalDetails: null,
  applicationRows: [],
};

class DemographicsBanner extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      demographicsMessages: require(`./translations/${props.locale}.js`),
    };
  }

  componentDidMount() {
    i18nLoader(this.props.locale, this.setState, this);
  }

  render() {
    if (this.state.load) {
      return (
        <I18nProvider locale={this.props.locale} messages={Object.assign(this.state.messages, this.state.demographicsMessages)}>
          <DemographicsBannerDisplay
            personName={this.props.personName}
            preferredFirstName={this.props.preferredFirstName}
            photo={this.props.photo}
            age={this.props.age}
            gender={this.props.gender}
            locale={this.props.locale}
            identifiers={this.props.identifiers}
            dateOfBirth={this.props.dateOfBirth}
            gestationalAge={this.props.gestationalAge}
            postMenstrualAge={this.props.postMenstrualAge}
            deceasedDate={this.props.deceasedDate}
            additionalDetails={this.props.additionalDetails}
            applicationRows={this.props.applicationRows}
          />
        </I18nProvider>
      );
    }

    return null;
  }
}

DemographicsBanner.propTypes = propTypes;
DemographicsBanner.defaultProps = defaultProps;

export default DemographicsBanner;
