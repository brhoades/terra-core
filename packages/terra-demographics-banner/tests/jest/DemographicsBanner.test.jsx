import React from 'react';
import DemographicsBanner from '../../src/DemographicsBanner';

describe('DemographicsBanner', () => {
  it('renders a blank banner properly', () => {
    const banner = shallow(<DemographicsBanner />);
    expect(banner).toMatchSnapshot();
  });

  it('renders a banner that has basic information', () => {
    const banner = shallow(
      <DemographicsBanner
        age="24"
        gender="F"
        personName="Terra Demographics Banner"
        dateOfBirth="March 30, 2017"
      />,
    );

    expect(banner).toMatchSnapshot();
  });

  it('renders the banner that contains additional information', () => {
    const banner = shallow(
      <DemographicsBanner
        photo={<img alt="My Cat" src="http://lorempixel.com/100/100/animals/7/" />}
        personName="Johnathon Doe"
        preferredFirstName="John"
        gender="Male"
        age="25 Years"
        dateOfBirth="May 9, 1993"
        additionalDetails={<span className="risk-score">5%</span>}
        identifiers={{ MRN: 12343, REA: '3JSDA' }}
        applicationRows={[{ Address: '1501 Walnut St. Apt #123', City: 'Kansas City MO' }, { ZIP: 64108 }]}
      />,
    );

    expect(banner).toMatchSnapshot();
  });

  it('renders the banner appropriately for a deceased person', () => {
    const banner = shallow(
      <DemographicsBanner
        age="24"
        gender="F"
        personName="Old Terra Demographics Banner"
        dateOfBirth="June 15, 2014"
        deceasedDate="March 30, 2017"
      />,
    );

    expect(banner).toMatchSnapshot();
  });

  it('renders the banner properly for a deceased person with additional application content', () => {
    const banner = shallow(
      <DemographicsBanner
        photo={<img alt="My Cat" src="http://lorempixel.com/100/100/animals/7/" />}
        personName="Johnathon Doe"
        preferredFirstName="John"
        gender="Male"
        age="25 Years"
        deceased="March 5, 2016"
        dateOfBirth="May 9, 1993"
        additionalDetails={<span className="risk-score">5%</span>}
        identifiers={{ MRN: 12343, REA: '3JSDA' }}
        applicationRows={[{ Address: '1501 Walnut St. Apt #123', City: 'Kansas City MO' }, { ZIP: 64108 }]}
      />,
    );

    expect(banner).toMatchSnapshot();
  });
});
