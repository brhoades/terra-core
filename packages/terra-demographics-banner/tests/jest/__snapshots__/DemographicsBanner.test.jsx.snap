exports[`test renders a banner that has basic information 1`] = `
<div>
  <section
    additionalDetails={null}
    age="24"
    applicationRows={Array []}
    className="terra-DemographicsBanner"
    dateOfBirth="March 30, 2017"
    deceasedDate={null}
    gender="F"
    gestationalAge={null}
    identifiers={Object {}}
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {
          "Terra.demographicsBanner.dateOfBirth": "DOB: <b>{dob}</b>",
          "Terra.demographicsBanner.deceased": "Deceased: <b>{deceasedDate}</b>",
          "Terra.demographicsBanner.gestationalAge": "GA: <b>{gestationalAge}</b>",
          "Terra.demographicsBanner.noDataProvided": "--",
          "Terra.demographicsBanner.postMenstrualAge": "PMA: <b>{postMenstrualAge}</b>",
        },
        "now": [Function],
        "textComponent": "span",
      }
    }
    locale="en"
    personName="Terra Demographics Banner"
    photo={null}
    postMenstrualAge={null}
    preferredFirstName={null}
    sex={null}>
    <div />
    <div
      className="terra-DemographicsBanner--content">
      <div
        className="terra-DemographicsBanner--row">
        <h1>
          Terra Demographics Banner
        </h1>
        <div
          className="terra-DemographicsBanner--application-content" />
      </div>
      <div
        className="terra-DemographicsBanner--row">
        <div
          className="terra-DemographicsBanner--person-details">
          <span>
            <b>
              24
            </b>
          </span>
          <span>
            <b>
              F
            </b>
          </span>
          <span>
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "DOB: <b>March 30, 2017</b>",
                }
              } />
          </span>
        </div>
        <div
          className="terra-DemographicsBanner--identifiers" />
      </div>
    </div>
  </section>
</div>
`;

exports[`test renders a blank banner properly 1`] = `
<div>
  <section
    additionalDetails={null}
    age={null}
    applicationRows={Array []}
    className="terra-DemographicsBanner"
    dateOfBirth={null}
    deceasedDate={null}
    gender={null}
    gestationalAge={null}
    identifiers={Object {}}
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {
          "Terra.demographicsBanner.dateOfBirth": "DOB: <b>{dob}</b>",
          "Terra.demographicsBanner.deceased": "Deceased: <b>{deceasedDate}</b>",
          "Terra.demographicsBanner.gestationalAge": "GA: <b>{gestationalAge}</b>",
          "Terra.demographicsBanner.noDataProvided": "--",
          "Terra.demographicsBanner.postMenstrualAge": "PMA: <b>{postMenstrualAge}</b>",
        },
        "now": [Function],
        "textComponent": "span",
      }
    }
    locale="en"
    personName={null}
    photo={null}
    postMenstrualAge={null}
    preferredFirstName={null}
    sex={null}>
    <div />
    <div
      className="terra-DemographicsBanner--content">
      <div
        className="terra-DemographicsBanner--row">
        <h1>
          <span
            dangerouslySetInnerHTML={
              Object {
                "__html": "--",
              }
            } />
        </h1>
        <div
          className="terra-DemographicsBanner--application-content" />
      </div>
      <div
        className="terra-DemographicsBanner--row">
        <div
          className="terra-DemographicsBanner--person-details">
          <span>
            <b>
              --
            </b>
          </span>
          <span>
            <b>
              --
            </b>
          </span>
          <span>
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "DOB: <b>--</b>",
                }
              } />
          </span>
        </div>
        <div
          className="terra-DemographicsBanner--identifiers" />
      </div>
    </div>
  </section>
</div>
`;

exports[`test renders the banner appropriately for a deceased person 1`] = `
<div>
  <section
    additionalDetails={null}
    age="24"
    applicationRows={Array []}
    className="terra-DemographicsBanner terra-DemographicsBanner--deceased"
    dateOfBirth="June 15, 2014"
    deceasedDate="March 30, 2017"
    gender="F"
    gestationalAge={null}
    identifiers={Object {}}
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {
          "Terra.demographicsBanner.dateOfBirth": "DOB: <b>{dob}</b>",
          "Terra.demographicsBanner.deceased": "Deceased: <b>{deceasedDate}</b>",
          "Terra.demographicsBanner.gestationalAge": "GA: <b>{gestationalAge}</b>",
          "Terra.demographicsBanner.noDataProvided": "--",
          "Terra.demographicsBanner.postMenstrualAge": "PMA: <b>{postMenstrualAge}</b>",
        },
        "now": [Function],
        "textComponent": "span",
      }
    }
    locale="en"
    personName="Old Terra Demographics Banner"
    photo={null}
    postMenstrualAge={null}
    preferredFirstName={null}
    sex={null}>
    <div />
    <div
      className="terra-DemographicsBanner--content">
      <div
        className="terra-DemographicsBanner--row">
        <h1>
          Old Terra Demographics Banner
        </h1>
        <div
          className="terra-DemographicsBanner--application-content" />
      </div>
      <div
        className="terra-DemographicsBanner--row">
        <div
          className="terra-DemographicsBanner--person-details">
          <span>
            <b>
              24
            </b>
          </span>
          <span>
            <b>
              F
            </b>
          </span>
          <span>
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "DOB: <b>June 15, 2014</b>",
                }
              } />
          </span>
          <span>
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Deceased: <b>March 30, 2017</b>",
                }
              } />
          </span>
        </div>
        <div
          className="terra-DemographicsBanner--identifiers" />
      </div>
    </div>
  </section>
</div>
`;

exports[`test renders the banner properly for a deceased person with additional application content 1`] = `
<div>
  <section
    additionalDetails={
      <span
        className="risk-score">
        5%
      </span>
    }
    age="25 Years"
    applicationRows={
      Array [
        Object {
          "Address": "1501 Walnut St. Apt #123",
          "City": "Kansas City MO",
        },
        Object {
          "ZIP": 64108,
        },
      ]
    }
    className="terra-DemographicsBanner"
    dateOfBirth="May 9, 1993"
    deceased="March 5, 2016"
    deceasedDate={null}
    gender="Male"
    gestationalAge={null}
    identifiers={
      Object {
        "MRN": 12343,
        "REA": "3JSDA",
      }
    }
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {
          "Terra.demographicsBanner.dateOfBirth": "DOB: <b>{dob}</b>",
          "Terra.demographicsBanner.deceased": "Deceased: <b>{deceasedDate}</b>",
          "Terra.demographicsBanner.gestationalAge": "GA: <b>{gestationalAge}</b>",
          "Terra.demographicsBanner.noDataProvided": "--",
          "Terra.demographicsBanner.postMenstrualAge": "PMA: <b>{postMenstrualAge}</b>",
        },
        "now": [Function],
        "textComponent": "span",
      }
    }
    locale="en"
    personName="Johnathon Doe"
    photo={
      <img
        alt="My Cat"
        src="http://lorempixel.com/100/100/animals/7/" />
    }
    postMenstrualAge={null}
    preferredFirstName="John"
    sex={null}>
    <div>
      <img
        alt="My Cat"
        src="http://lorempixel.com/100/100/animals/7/" />
    </div>
    <div
      className="terra-DemographicsBanner--content">
      <div
        className="terra-DemographicsBanner--row">
        <h1>
          Johnathon Doe
          <span
            className="terra-DemographicsBanner--preferred-first-name">
            John
          </span>
        </h1>
        <div
          className="terra-DemographicsBanner--application-content">
          <span
            className="risk-score">
            5%
          </span>
        </div>
      </div>
      <div
        className="terra-DemographicsBanner--row">
        <div
          className="terra-DemographicsBanner--person-details">
          <span>
            <b>
              25 Years
            </b>
          </span>
          <span>
            <b>
              Male
            </b>
          </span>
          <span>
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "DOB: <b>May 9, 1993</b>",
                }
              } />
          </span>
        </div>
        <div
          className="terra-DemographicsBanner--identifiers">
          <span>
            MRN
            : 
            <b>
              12343
            </b>
          </span>
          <span>
            REA
            : 
            <b>
              3JSDA
            </b>
          </span>
        </div>
      </div>
      <div
        className="terra-DemographicsBanner--row">
        <span>
          Address
          : 
          <b>
            1501 Walnut St. Apt #123
          </b>
        </span>
        <span>
          City
          : 
          <b>
            Kansas City MO
          </b>
        </span>
      </div>
      <div
        className="terra-DemographicsBanner--row">
        <span>
          ZIP
          : 
          <b>
            64108
          </b>
        </span>
      </div>
    </div>
  </section>
</div>
`;

exports[`test renders the banner that contains additional information 1`] = `
<div>
  <section
    additionalDetails={
      <span
        className="risk-score">
        5%
      </span>
    }
    age="25 Years"
    applicationRows={
      Array [
        Object {
          "Address": "1501 Walnut St. Apt #123",
          "City": "Kansas City MO",
        },
        Object {
          "ZIP": 64108,
        },
      ]
    }
    className="terra-DemographicsBanner"
    dateOfBirth="May 9, 1993"
    deceasedDate={null}
    gender="Male"
    gestationalAge={null}
    identifiers={
      Object {
        "MRN": 12343,
        "REA": "3JSDA",
      }
    }
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": Object {
          "Terra.demographicsBanner.dateOfBirth": "DOB: <b>{dob}</b>",
          "Terra.demographicsBanner.deceased": "Deceased: <b>{deceasedDate}</b>",
          "Terra.demographicsBanner.gestationalAge": "GA: <b>{gestationalAge}</b>",
          "Terra.demographicsBanner.noDataProvided": "--",
          "Terra.demographicsBanner.postMenstrualAge": "PMA: <b>{postMenstrualAge}</b>",
        },
        "now": [Function],
        "textComponent": "span",
      }
    }
    locale="en"
    personName="Johnathon Doe"
    photo={
      <img
        alt="My Cat"
        src="http://lorempixel.com/100/100/animals/7/" />
    }
    postMenstrualAge={null}
    preferredFirstName="John"
    sex={null}>
    <div>
      <img
        alt="My Cat"
        src="http://lorempixel.com/100/100/animals/7/" />
    </div>
    <div
      className="terra-DemographicsBanner--content">
      <div
        className="terra-DemographicsBanner--row">
        <h1>
          Johnathon Doe
          <span
            className="terra-DemographicsBanner--preferred-first-name">
            John
          </span>
        </h1>
        <div
          className="terra-DemographicsBanner--application-content">
          <span
            className="risk-score">
            5%
          </span>
        </div>
      </div>
      <div
        className="terra-DemographicsBanner--row">
        <div
          className="terra-DemographicsBanner--person-details">
          <span>
            <b>
              25 Years
            </b>
          </span>
          <span>
            <b>
              Male
            </b>
          </span>
          <span>
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "DOB: <b>May 9, 1993</b>",
                }
              } />
          </span>
        </div>
        <div
          className="terra-DemographicsBanner--identifiers">
          <span>
            MRN
            : 
            <b>
              12343
            </b>
          </span>
          <span>
            REA
            : 
            <b>
              3JSDA
            </b>
          </span>
        </div>
      </div>
      <div
        className="terra-DemographicsBanner--row">
        <span>
          Address
          : 
          <b>
            1501 Walnut St. Apt #123
          </b>
        </span>
        <span>
          City
          : 
          <b>
            Kansas City MO
          </b>
        </span>
      </div>
      <div
        className="terra-DemographicsBanner--row">
        <span>
          ZIP
          : 
          <b>
            64108
          </b>
        </span>
      </div>
    </div>
  </section>
</div>
`;
